src/libraries/Directory.Build.targets

```diff
@@ -33,6 +33,12 @@
          SYSLIB0036: Regex.CompileToAssembly
     -->
     <NoWarn Condition="'$(IsPartialFacadeAssembly)' == 'true'">$(NoWarn);SYSLIB0003;SYSLIB0004;SYSLIB0015;SYSLIB0017;SYSLIB0021;SYSLIB0022;SYSLIB0023;SYSLIB0025;SYSLIB0032;SYSLIB0036</NoWarn>
+
+    <!-- Ignore analyzers that recommend APIs introduced in .NET Core when targeting frameworks that lack those APIs
+         to avoid issues with multitargeting.
+    -->
+    <NoWarn Condition="$(TargetFrameworks.Contains('NetFramework')) or $(TargetFrameworks.Contains('netstandard'))">$(NoWarn);CA1510;CA1511;CA1512;CA1513</NoWarn>
```

Comment:
TargetFrameworks.Contains('NetFramework')) is never true as .NET Framework tfms don't contain such a string in their tfm alias. Instead you want to use TargetFrameworkIdentifier. Using TFI guarantees that the logic works, irrespective of the alias which can be any value.

```
<NoWarn Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' or '$(TargetFrameworkIdentifier)' == '.NETStandard'">$(NoWarn);CA1510;CA1511;CA1512;CA1513</NoWarn>
```

Reply:
I see, I thought you wanted to set the NoWarn for .NET Framework and .NET Standard builds. Still, `$(TargetFrameworks.Contains('NetFramework'))` is wrong, we don't have such a TFM. You probably conditioned on the msbuild property instead of the property value.

We usually use `net4` to identify .NET Framework builds in the TargetFrameworks string:

```
<NoWarn Condition="$(TargetFrameworks.Contains('net4')) or $(TargetFrameworks.Contains('netstandard'))">$(NoWarn);CA1510;CA1511;CA1512;CA1513</NoWarn>
```